pkg/reflector/download.go

```diff
@@ -33,17 +37,28 @@ type S3Downloader struct {
 
 func (d *S3Downloader) DownloadTo(w io.Writer) (n int64, err error) {
 	client, err := d.getS3Client()
+	downloader := manager.NewDownloader(client, func(d *manager.Downloader) {
+		d.PartSize = 64 * 1024 * 1024 // 64MB per part
+		d.Concurrency = 5
+	})
 	if err != nil {
 		return -1, err
 	}
+
+	tmpdir := os.TempDir()
+	file, err := os.CreateTemp(tmpdir, d.Key)
+	if err != nil {
+		return -1, errors.Wrap(err, "download snapshot into disk")
+	}
+	defer os.Remove(file.Name())
+
 	start := time.Now()
-	defer func() {
-		stats.Observe("snapshot_download_time", time.Now().Sub(start))
-	}()
-	obj, err := client.GetObject(&s3.GetObjectInput{
+	numBytes, err := downloader.Download(context.TODO(), file, &s3.GetObjectInput{
 		Bucket: aws.String(d.Bucket),
 		Key:    aws.String(d.Key),
 	})
+	stats.Observe("snapshot_download_time", time.Now().Sub(start))
```

Comment:
Do you know the default string output of `time.Now().Sub(start)`? If it's ms then should you change the title of the metric to include `_ms` or whichever unit it is?